cpp/arcticdb/version/version_core.hpp

```diff
@@ -96,8 +96,25 @@ VersionedItem compact_incomplete_impl(
     bool append,
     bool convert_int_to_float,
     bool via_iteration,
-    bool sparsify);
-
+    bool sparsify,
+    const WriteOptions& write_options);
+
+std::tuple<std::shared_ptr<PipelineContext>, ReadQuery, size_t, std::optional<size_t>> get_pre_defragmentation_info(
```

Comment:
Please write some docs. Returning a `ReadQuery` is most unusual.

The last two items effectively have the same type. Should really return a `struct` with each field properly named.

Reply:
1. `ReadQuery` is required by `Clause`, (`clause.hpp`)
2. It is arguable. If in doubt,  meaning of fields can be found in the caller of `get_pre_defragmentation_info`, as the variables are named properly over there.

Reply:
Is the `ReadQuery` returned empty?